### PR TITLE
Fix RNA-seq unittest (resolves #313)

### LINKED ISSUE #313: RNASeq-CGL pipeline failing unit test on Jenkins
Failure observed on Jenkins build 281 in #265.

```
=================================== FAILURES ===================================
_______________________________ test_rnaseq_cgl ________________________________

tmpdir = local('/tmp/pytest-of-jenkins/pytest-13/test_rnaseq_cgl0')

    def test_rnaseq_cgl(tmpdir):
        work_dir = str(tmpdir)
        create_config_and_manifest(work_dir)
        subdir = '/mnt/ephemeral/toil-scripts/rnaseq/'
>       os.makedirs(os.path.join(subdir, 'workDir'))

src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '/mnt/ephemeral/toil-scripts/rnaseq/workDir', mode = 511

    def makedirs(name, mode=0777):
        """makedirs(path [, mode=0777])
    
        Super-mkdir; create a leaf directory and all intermediate ones.
        Works like mkdir, except that any intermediate path segment (not
        just the rightmost) will be created if it does not exist.  This is
        recursive.
    
        """
        head, tail = path.split(name)
        if not tail:
            head, tail = path.split(head)
        if head and tail and not path.exists(head):
            try:
                makedirs(head, mode)
            except OSError, e:
                # be happy if someone already created the path
                if e.errno != errno.EEXIST:
                    raise
            if tail == curdir:           # xxx/newdir/. exists if xxx/newdir exists
                return
>       mkdir(name, mode)
E       OSError: [Errno 17] File exists: '/mnt/ephemeral/toil-scripts/rnaseq/workDir'

venv/lib/python2.7/os.py:157: OSError
-- generated xml file: /mnt/ephemeral/workspace/toil-scripts/test-report.xml ---
==================== 1 failed, 18 passed in 302.13 seconds =====================
make: *** [test] Error 1
Build step 'Execute shell' marked build as failure
Recording test results
Setting status of 43793ab48096367e9ce95c53d63b2f618dc58352 to FAILURE with url http://jenkins.cgcloud.info/job/toil-scripts/281/ and message: 'Build finished. 19 tests run, 0 skipped, 1 failed.'
Using context: jenkins
Finished: FAILURE
```

### PR DESCRIPTION
**Pull request by jvivian**

### CODE REVIEW

**Comment by jvivian** [comment]
@hannes-ucsc @almussel — Ready for review

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py [hunk 1]
```diff
@@ -2,27 +2,30 @@
...
 import textwrap
+from uuid import uuid4
 
 from boto.s3.connection import S3Connection, Bucket, Key
 
 
 def test_rnaseq_cgl(tmpdir):
-    work_dir = str(tmpdir)
-    create_config_and_manifest(work_dir)
+    tmp_dir = str(tmpdir)
```
Comment (fnothaft) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: There are several throwaway directories floating around here (`tmpdir`, `tmp_dir`, and `work_dir`). Can you document what each one of them does or how they are different?

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py [hunk 2]
```diff
@@ -2,27 +2,30 @@
...
+    create_config_and_manifest(tmp_dir)
     subdir = '/mnt/ephemeral/toil-scripts/rnaseq/'
-    os.makedirs(os.path.join(subdir, 'workDir'))
+    work_dir = os.path.join(subdir, str(uuid4()))
```
Comment (fnothaft) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: Factor out L14–L16 to a `make_temp_dir` function?

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py (hunk 1 above)
Comment (hannes-ucsc) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: Also, what type is tmpdir such that you have to convert it to a string?

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py [hunk 3]
```diff
@@ -2,27 +2,30 @@
...
+    if not os.path.exists(work_dir):
```
Comment (hannes-ucsc) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: BPL has mkdir_p

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py (hunk 2 above)
Comment (fnothaft) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: Also, see comments RE: L13 in #315.

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py (hunk 1 above)
Comment (jvivian) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: > Also, what type is tmpdir such that you have to convert it to a string?

PyTest uses a [py.path.local object](http://py.readthedocs.io/en/latest/path.html)

File: src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py (hunk 3 above)
Comment (jvivian) [inline src/toil_scripts/rnaseq_cgl/test/test_rnaseq_cgl.py]: Thanks, forgot about that.

**Comment by jvivian** [comment]
Addressed comments except the hardcoded mount directory.  Don't know if we want to delay this PR until we figure that out.

**Comment by fnothaft** [comment]
> Addressed comments except the hardcoded mount directory. Don't know if we want to delay this PR until we figure that out.

I would rather not delay the PR on that.

**Comment by almussel** [comment]
+1 I would agree with @fnothaft since we're waiting on this for #265.

**Comment by jvivian** [comment]
@hannes-ucsc — Dealt with comments aside from #315 which I've self-assigned.

**Comment by hannes-ucsc** [comment]
+1 from me. @fnothaft?

**Comment by jvivian** [comment]
Once this is merged I can submit a PR for #315

**Pull request merged**